### ISSUE TITLE
Create a fork/branch to help test remote push notifications

### ISSUE THREAD

**Issue by hboon** [AlphaWallet/alpha-wallet-android#3110]
There are 2 options:

A1. Create a new branch and add the functionality
B1. Create a new branch and strip down the app to 0

Either way, this is not meant to be merged into `master`. Our backend developers will build this branch themselves for testing

Functionality:

A. Use Firebase
B. TextBox to support entering a wallet address
C. A button to request for push notification permissions (should always re-try if tapped)
D. That button should also register with our backend API. Ought to be simple POST-type call (details pending) passing in the address+device push token
E. When a remote push is received, display the entire JSON payload. The remote push notification is supposed to be a silent notification — so when we receive it while the app is active: display the JSON. If the app is inactive: show a pop up notification with the JSON. Either way, we should indicate if it is from a silent push or not.
F. A button top undo (C) by making a different call

**Comment by hboon** [comment]
cc @abcfy2 @foxgem

**Comment by justindg** [comment]
@hboon @abcfy2 @foxgem 

I created a branch called `notification-test`. I replaced the Activity tab with the notification test UI.

I've been trying to test using the FCM console - I am able to receive notifications and display payload, however, through the console I am not able to accomplish two things:
- Receive silent notifications
- Obtain notification payload when app is on background

I think these need to be implemented from our own server. Please let me know if you know any way to test it. 

Also do let me know if you have any questions regarding the branch. The relevant files are:

`NotificationTestFragment`  - mostly controls the UI and flow
`NotificationTestViewModel` - business logic handling
`AlphaWalletFirebaseMessagingService` - this is where the notification payload is processed
`NotificationTestService` - scaffolding for calling custom backend APIs

**Comment by hboon** [comment]
Thanks @justindg. Just to anticipate things a bit — would you help include a pointer to how to build the project? At least starting point :)

**Comment by justindg** [comment]
Latest Android Studio version: https://developer.android.com/studio  

1. In Android Studio, `File` > `Open` > <path>/alpha-wallet-android
2. Wait for gradle to fully sync and download the dependencies

Preparation:
1. In Android Studio, go to `Tools` > `SDK Manager`
2. Under `SDK Platforms`, select the Android versions you wish to download (Android 12.0 or 13.0 is recommended)
3. Under `SDK Tools`, select the following items:
     - `Android SDK Build-Tools`
     - `NDK`
     - `Android SDK Command-line Tools`
     - `CMake`
     - `Android Emulator`
     - `Android SDK Platform-Tools`
     - `Intel x86 Emulator Accelerator`
4. Click `Apply`, accept the licenses and wait for everything to download.

To test using a real device:

1. Make sure your device's `Developer options` are enabled. Go to `Settings` > `System` > `About phone` > Tap `Build number` around 5-7 times until you see a prompt. Note: this may vary depending on the phone manufacturer
2. Once enabled, go to `Settings` > `Developer options` > Enable `USB Debugging`
3. Once this is done, you should see something like this in the Android Studio toolbar
![image](https://user-images.githubusercontent.com/17334718/222605116-8c8129fb-cc24-47e3-832b-c79224bbda5b.png)
4. Click on the green play button to run the app. AlphaWallet should shortly launch on your phone.

To test using an emulator:
1. In Android Studio, go to `Tools` > `Device Manager`
2. Click `Create device` on the prompt and follow the instructions.
3. You can now launch the emulator to run the app.

**Comment by hboon** [comment]
@justindg how is this one going?

**Comment by justindg** [comment]
> @justindg how is this one going?

The testing was successful, I was able to receive the notifications. What are the next steps for this?

**Comment by hboon** [comment]
Summary for discussion on Discord: basically change it to silent/data push and we can close this and move to next step for push notifications.

**Comment by hboon** [comment]
@justindg can close this since we moved on to #3152
